File: C/Colleen.c
#include <stdio.h>

/*
	Global scope
*/

void printer()
{
	char *str = "#include <stdio.h>%1$c%1$c/*%1$c%2$cGlobal scope%1$c*/%1$c%1$cvoid printer()%1$c{%1$c%2$cchar *str = %4$c%3$s%4$c;%1$c%2$cprintf(str, 10, 9, str, 34);%1$c}%1$c%1$cint main()%1$c{%1$c%2$c/*%1$c%2$c%2$cLocal scope%1$c%2$c*/%1$c%2$cprinter();%1$c%2$creturn (0);%1$c}%1$c";
	printf(str, 10, 9, str, 34);
}

int main()
{
	/*
		Local scope
	*/
	printer();
	return (0);
}

File: python/Colleen.py
# Global scope comment

def printer():
	str = "# Global scope comment\n\ndef printer():\n\tstr = %r\n\tprint(str %% (str,))\n\nif __name__ == '__main__':\n\t# Local scope of the main function\n\tprinter()"
	print(str % (str,))

if __name__ == '__main__':
	# Local scope of the main function
	printer()
